### PR TITLE
ENH: Add more table init benchmarks from taldcroft

### LINKED ISSUE #99: Add benchmarks for table creation from lists
Relevant to https://github.com/astropy/astropy/pull/10636 and the regression introduced in https://github.com/astropy/astropy/pull/9048. The notebooks in 10636 can form the basis for testing.

### PR DESCRIPTION
**Pull request by pllim**
Fix #99 

I only ported over the cells with `timeit` in https://gist.github.com/taldcroft/2fa546ecb13ebe4a7ebc33ffa5dbb016 . cc @mhvk 

The `setup` has a lot of copies of large arrays. If this will present memory problem, please let me know how to mitigate the problem. Thanks!

💯

### CODE REVIEW

File: benchmarks/table.py [hunk 1]
```diff
@@ -182,3 +182,70 @@ def setup(self):
...
+        self.data_str_masked_3d = self.data_str_3d.copy()
+
+        self.data_str_masked_1d[-1] = np.ma.masked
+        self.data_str_masked_2d[-1][-1] = np.ma.masked
+        self.data_str_masked_3d[-1][-1][-1] = np.ma.masked
+
+    def time_init_int_1d(self):
+        Table([self.data_int_1d])
+
+    def time_init_int_2d(self):
```
Comment (taldcroft) [inline benchmarks/table.py:220]: Cut.

File: benchmarks/table.py [hunk 2]
```diff
@@ -182,3 +182,70 @@ def setup(self):
...
+        Table([self.data_int_2d])
+
+    def time_init_int_3d(self):
+        Table([self.data_int_3d])
+
+    def time_init_int_masked_1d(self):
+        Table([self.data_int_masked_1d])
+
+    def time_init_int_masked_2d(self):
```
Comment (taldcroft) [inline benchmarks/table.py:229]: Cut.

File: benchmarks/table.py [hunk 3]
```diff
@@ -182,3 +182,70 @@ def setup(self):
...
+        Table([self.data_int_masked_2d])
+
+    def time_init_int_masked_3d(self):
+        Table([self.data_int_masked_3d])
+
+    def time_init_str_1d(self):
+        Table([self.data_str_1d])
+
+    def time_init_str_2d(self):
```
Comment (taldcroft) [inline benchmarks/table.py:238]: Cut.

File: benchmarks/table.py [hunk 4]
```diff
@@ -182,3 +182,70 @@ def setup(self):
...
+        Table([self.data_str_2d])
+
+    def time_init_str_3d(self):
```
Comment (taldcroft) [inline benchmarks/table.py:241]: Cut.

File: benchmarks/table.py [hunk 5]
```diff
@@ -182,3 +182,70 @@ def setup(self):
...
+        Table([self.data_str_3d])
+
+    def time_init_str_masked_1d(self):
+        Table([self.data_str_masked_1d])
+
+    def time_init_str_masked_2d(self):
```
Comment (taldcroft) [inline benchmarks/table.py:247]: Cut.

File: benchmarks/table.py [hunk 6]
```diff
@@ -182,3 +182,70 @@ def setup(self):
...
+        Table([self.data_str_masked_2d])
+
+    def time_init_str_masked_3d(self):
```
Comment (taldcroft) [inline benchmarks/table.py:250]: Cut.

**Comment by pllim** [comment]
Thanks for the review, @taldcroft ! I have applied all your comments.

**Pull request merged**